src/app/users/page.tsx

```diff
@@ -1,9 +1,73 @@
-import React from 'react'
+"use client";
+import axios from "axios";
+import { useEffect, useState } from "react";
 
-const User = () => {
-  return (
-    <div> User</div>
-  )
+interface User {
+  id: number;
+  firstName: string;
+  lastName: string;
+  email: string;
+  fatherName: string;
+  motherName: string;
+  address: string;
+  pincode: string;
+  country: string;
 }
 
-export default User
\ No newline at end of file
+const Users = () => {
+  const [userData, setUserData] = useState<User[]>([]);
+
+  const getUsers = async () => {
+    try {
+      const res = await axios.get<User[]>("/api/getuser");
+      setUserData(res.data);
+      console.log(res, "res");
+    } catch (error) {
+      console.error("Error fetching users:", error);
+    }
+  };
+
+  useEffect(() => {
+    getUsers();
+  }, []);
+
+  return (
```

Comment:
this page is not server side, as asked please use only server side rendering, read next js docs what to do for server side rendering